### DISCUSSION TITLE
Tutorial did not work as expected (unmapped fields)

### DISCUSSION THREAD

**Discussion by snow6oy** [graphql-java/graphql-java#3482] [Q&A] [answered]
Steps from the guide that I followed on [spring.io](https://spring.io/guides/gs/graphql-server/). It looks like the same tutorial is on graphql-java.com too.

Took the option to [generate the zip](https://start.spring.io/) with these parameters

- Project: Maven
- Language: Java
- Springboot: 3.2.3
- Project Metadata: accept all defaults.
- Packaging: JAR
- Java: 17
- Dependencies: Spring for GraphQL and Spring Web

Unzip and cd into workspace, then add these files as cut-n-paste from the Tutorial.

- src/main/java/com/example/demo/BookController.java
- src/main/java/com/example/demo/Book.java
- src/main/java/com/example/demo/Author.java
- src/main/resources/graphql/schema.graphqls 

and enable GraphiQL. Then I run 'mvn package' and see a warning

```
2024-02-28T15:43:10.217Z  INFO 84590 --- [           main] o.s.b.a.g.GraphQlAutoConfiguration       : GraphQL schema inspection:
        Unmapped fields: {Query=[bookById]}
        Unmapped registrations: {}
        Skipped types: []
```

and so then at runtime the query returns 'null' data. In my limited understanding, Spring is saying that it cannot match between what is written to the @QueryMapping annotation

```
@Controller
public class BookController {
    @QueryMapping
    public Book bookById(@Argument String id) {
        return Book.getById(id);
    }

```
And what is defined in the schema

```
type Query {
    bookById(id: ID): Book
}

```
But as this is my first attempt at Spring GraphQL I am shooting in the dark :-) 
Can someone with experience please compare the annotated java and the schema. Or perhaps how to run a debugger to get more detail.

Thanks

**Comment by snow6oy** [comment, marked as answer]
There was a conflict with the package names. It does work after all :smile:

**Reply by dondonz** [reply]
Was there a problem with the package names in the tutorial?

**Reply by dondonz** [reply]
At the end of the tutorial there is a link to the GitHub repository so you can compare side by side https://github.com/graphql-java/tutorials
